x/interchainquery/keeper/msg_submit_query_response_test.go

```diff
@@ -91,33 +94,64 @@ func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_UnknownId() {
 	s.Require().True(found)
 }
 
-func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_ExceededTtl() {
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_Timeout_RejectQuery() {
 	tc := s.SetupMsgSubmitQueryResponse()
 
-	// Remove key from the query type so to bypass the VerifyKeyProof function
-	tc.query.QueryType = strings.ReplaceAll(tc.query.QueryType, "key", "")
+	// set timeout to be expired and set the policy to reject
+	tc.query.TimeoutTimestamp = uint64(1)
+	tc.query.TimeoutPolicy = types.TimeoutPolicy_REJECT_QUERY_RESPONSE
+	s.App.InterchainqueryKeeper.SetQuery(s.Ctx, tc.query)
+
+	_, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
+	s.Require().NoError(err)
 
-	// set timeout to be expired
+	// check that the query original query was deleted,
+	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx, tc.query.Id)
+	s.Require().False(found, "original query should be removed")
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_Timeout_RetryQuery() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	// set timeout to be expired and set the policy to retry
 	tc.query.TimeoutTimestamp = uint64(1)
+	tc.query.TimeoutPolicy = types.TimeoutPolicy_RETRY_QUERY_REQUEST
 	s.App.InterchainqueryKeeper.SetQuery(s.Ctx, tc.query)
 
-	resp, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
+	_, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
 	s.Require().NoError(err)
-	s.Require().NotNil(resp)
 
-	// check that the query was not deleted (since the query timed out)
+	// check that the query original query was deleted,
+	// but that a new one was created for the retry
 	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx, tc.query.Id)
-	s.Require().True(found)
+	s.Require().False(found, "original query should be removed")
+	s.Require().Len(s.App.InterchainqueryKeeper.AllQueries(s.Ctx), 1, "there should be one new query")
```

Comment:
Can we also check the attributes of the new query are what we expect them to be?

Reply:
done